cmd/proxy/main.go

```diff
@@ -234,7 +284,29 @@ func (p *Proxy) run(o *ProxyRunOptions) error {
 		return fmt.Errorf("failed to validate server options with %v", err)
 	}
 	ctx, cancel := context.WithCancel(context.Background())
-	server := agentserver.NewProxyServer()
+	defer cancel()
+
+	var k8sClient *kubernetes.Clientset
+	if o.authenticateAgent {
+		config, err := clientcmd.BuildConfigFromFlags("", o.kubeconfigPath)
```

Comment:
Maybe add a validation for kubeconfig to fail in validate if it is empty/non existent?

Reply:
Empty string is a valid case, since clientcmd.BuildConfigFromFlags take care for it, however, I added a validation case, to check the file exists if kubeconfigPath is provided